reactor-core/src/main/java/reactor/core/publisher/MonoPeekTerminal.java

```diff
@@ -378,12 +343,14 @@ public T poll() {
 
 		@Override
 		public boolean isEmpty() {
-			return queueSubscription.isEmpty();
+			return queueSubscription == null || queueSubscription.isEmpty();
 		}
 
 		@Override
 		public void clear() {
-			queueSubscription.clear();
+			if (queueSubscription != null) {
```

Comment:
These kind of checks are usually redundant as they are called only if fuseable. Wouldn't be covered by normal test coverage.

Reply:
I wanted to get rid of the pesky nullability warnings from static analysis... maybe an `assert queueSubscription != null` would be better overhead-wise?

Reply:
actually, isn't it safer to keep these checks, especially as the number of hybrid `Subscribers` that are instantiated for both `Fuseable` and non-fuseable versions grow? I expect the JVM to optimize these checks out at runtime anyway.

Reply:
right. I was thinking about discard scenarios where we can call `Queue::clear` by default, but these should be managed by the original subscription anyway...